lib/ecto/association.ex

```diff
@@ -282,6 +284,11 @@ defmodule Ecto.Association.Has do
                            "option, the model should not be passed as second argument"
     end
 
+    if opts[:dependent] && not opts[:dependent] in @dependent_opts do
```

Comment:
In theory the default is `:nothing` (if we want to mirror with the `on_delete` from migrations.